test/Table_Tests/src/IO/Delimited_Read_Spec.enso

```diff
@@ -225,13 +225,53 @@ spec =
 
         Test.specify "should fail in presence of a mismatched quote" <|
             [Problem_Behavior.Report_Error, Problem_Behavior.Report_Warning, Problem_Behavior.Ignore].each pb->
-                r1 = Data.read (enso_project.data / "mismatched_quote.csv") (Delimited "," headers=True value_formatter=Nothing) on_problems=pb
+                format = (Delimited "," headers=True value_formatter=Nothing)
+                r1 = Data.read (enso_project.data / "mismatched_quote.csv") format on_problems=pb
                 r1.should_fail_with File_Error.Corrupted_Format
                 r1.catch.to_display_text.should_contain "quote has been opened but never closed"
+                r1.catch.to_display_text.should_contain '["\n7,8,9\n]'
 
-                r2 = Data.read (enso_project.data / "mismatched_quote2.csv") (Delimited "," headers=True value_formatter=Nothing) on_problems=pb
+                r2 = Data.read (enso_project.data / "mismatched_quote2.csv") format on_problems=pb
                 r2.should_fail_with File_Error.Corrupted_Format
 
+                format3 = format . with_quotes quote_escape="\"
+                r3 = Data.read (enso_project.data / "mismatched_quote3.csv") format3 on_problems=pb
+                r3.should_fail_with File_Error.Corrupted_Format
+
+                r4 = Data.read (enso_project.data / "mismatched_quote4.csv") format on_problems=pb
+                r4.should_fail_with File_Error.Corrupted_Format
+
+                r5 = Data.read (enso_project.data / "mismatched_quote5.csv") format on_problems=pb
+                r5.should_fail_with File_Error.Corrupted_Format
+
+                f6 = enso_project.data / "transient" / "mismatched_quote3.csv"
+                # There is no trailing newline compared to regular `mismatched_quote3.csv`.
+                contents = 'a,b,c\n1,2,3\nabc,def,"g h i\\"'
+                contents.write f6 on_existing_file=Existing_File_Behavior.Overwrite
+                r6 = Data.read f6 format3 on_problems=pb
+                r6.should_fail_with File_Error.Corrupted_Format
+                f6.delete
```

Comment:
One more test that is failing.

So my hunch was right - the logic for finding mismatched quotes wasn't quite right. The tests were kind of passing by accident - in `mismatched_quote3.csv` we had `"g h i\"<newline>` loaded as the last cell - so the check saw that the last character is not a quote but in fact a newline.

Thus I added this test that does not have a newline at EOF and now it is failing - as in - it is actually loading a table without any errors.

Reply:
I think we need to properly parse the string handling the escape sequences and counting opened and closed quotes to figure out this. 

Probably only opening a quote if it opens at the beginning and closing it if any other (unescaped) quote character shows up. If we encounter further non-whitespace characters afterwards, we may want to report some other error like 'Unexpected Characters in cell XYZ'.

cc: @jdunkerley does that specification for amended mismatched quote handling sound good to you? It seems to me like the way to make these additional tests pass. The approach we have currently is not very robust and will break in some fun ways.

Second question: do we want to handle it right now or schedule it for later? As I know this kind of showed up as a side-product of the Table.read problem handling revision, I'm not sure how you think we should prioritize this. I think it should take about one more day (maybe ~half) to implement this new idea for quote handling which handles escapes (the algorithm is relatively simple, may probably need to add a few more test cases to make sure it is robust).

Reply:
| Probably only opening a quote if it opens at the beginning and closing it if any other (unescaped) quote character shows up.
This sounds correct. And yes I think that is the most robust approach is to report unexpected characters.

Lets ticket it up but probably leave for now.

Reply:
Added as ticket #5839